desc/profiles.py

```diff
@@ -687,6 +692,88 @@ def from_values(cls, x, y, order=6, rcond=None, w=None, sym="auto", name=""):
         return cls(params, sym=sym, name=name)
 
 
+class PowerLawProfile(_Profile):
+    """Profile represented by a power law.
+
+    f(x) = a[0]*(1 - x**a[1])**a[2]
+
+    Parameters
+    ----------
+    params: array-like
+        Coefficients of the power law series. Must be an array of size 3.
+        Assumed to be zero if not specified.
+    name : str
+        Name of the profile.
+
+    """
+
+    _io_attrs_ = _Profile._io_attrs_
+
+    def __init__(self, params=None, name=""):
+        super().__init__(name)
+
+        if params is None:
+            params = [0, 0, 0]
+        self._params = np.atleast_1d(params)
+
+        errorif(
+            self._params.size != 3, ValueError, "params must be an array of size 3."
+        )
+
+    @property
+    def params(self):
+        """ndarray: Parameter values."""
+        return self._params
+
+    @params.setter
+    def params(self, new):
+        new = jnp.atleast_1d(jnp.asarray(new))
+        if new.size == 3:
+            self._params = jnp.asarray(new)
+        else:
+            raise ValueError(f"params should be an array of size 3, got {len(new)}.")
+
+    def compute(self, grid, params=None, dr=0, dt=0, dz=0):
+        """Compute values of profile at specified nodes.
+
+        Parameters
+        ----------
+        grid : Grid
+            Locations to compute values at.
+        params : array-like
+            Power law coefficients to use. Must be an array of size 3.
+            If not given, uses the values given by the params attribute.
+        dr, dt, dz : int
+            Derivative order in rho, theta, zeta.
+
+        Returns
+        -------
+        values : ndarray
+            Values of the profile or its derivative at the points specified.
+
+        """
+        if params is None:
+            params = self.params
+        if (dt != 0) or (dz != 0):
+            return jnp.zeros(grid.num_nodes)
+        a, b, c = params
+        r = grid.nodes[:, 0]
+        if dr == 0:
+            f = a * (1 - r**b) ** c
+        # df/dr = inf at rho = 1 for all dr > 0
```

Comment:
isnt this only true if b,c are floats <dr?

what is the use case for b,c being floats rather than ints?

my worry is also that if b<1 than the derivative blows up at r=0 as well.

Reply:
True, depending on the parameter values it may or may not blow up at $\rho=0$ and/or $\rho=1$. 

My personal use case is to use an integer `b` = 1 or 2, and `c` as a float. But if these numbers are optimizable params they all need to be floats. 

Options:

1. Add a check to make sure `b >= 1`. But there isn't much we can do about `c`. 
2. We don't need to merge this into the master branch. But I coded it up anyways so I figured I would share it.

Reply:
My question is, do they all need to be optimizable? We could just make `b`,`c` constant integers and leave `a` optimizable?

Along a similar vein, VMEC has a bunch of other profiles types that might be worth looking at: https://github.com/jonathanschilling/educational_VMEC/blob/master/src/profile_functions.f

I have some VMEC input files that use the atan and two-lorentz profiles that would be good to try in DESC.